test/parallel/test-stream-readable-async-iterators.js

```diff
@@ -396,6 +396,26 @@ async function tests() {
     }
   }
 
+  console.log('readable side of a transform stream pushes null');
+  {
+    const transform = new Transform({
+      objectMode: true,
+      transform: common.mustNotCall()
+    });
+    transform.push(0);
+    transform.push(1);
+    transform.push(null);
+    const mustReach = common.mustCall();
+    const iter = transform[Symbol.asyncIterator]();
+    assert.strictEqual((await iter.next()).value, 0);
+
+    for await (const d of iter) {
+      assert.strictEqual(d, 1);
```

Comment:
Nit: this test does not test that the loop body is actually reached.

Reply:
I realised what I did wrong the null push should be in a nextTick